### PR TITLE
Improve description and examples of Performance/UnneededSort

### PR DESCRIPTION
**Pull request by parkerfinch**
Add a more thorough description and more exhaustive examples to Performance/UnneededSort.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/performance/unneeded_sort.rb [hunk 1]
```diff
@@ -3,17 +3,52 @@
...
+      # element and `Enumerable#max` instead of sorting and taking the
+      # last element. Similarly, `Enumerable#min_by` and
+      # `Enumerable#max_by` can replace `Enumerable#sort_by` calls
+      # after which only the first or last element is used.
       #
       # @example
       #   # bad
       #   [].sort.first
-      #   [].sort_by(&:length).last
+      #   [].sort[0]
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/unneeded_sort.rb]: You might also put some elements in those brackets or replace them with `arr`, `collection` or something like that. Looks odd to be sorting an empty array literal. ;-)

Comment (parkerfinch) [reply lib/rubocop/cop/performance/unneeded_sort.rb]: Done! Thanks for all your patience and advice on this!

Comment (bbatsov) [reply lib/rubocop/cop/performance/unneeded_sort.rb]: You're welcome!

**Comment by bbatsov** [comment]
Nice!

**Pull request merged**